src/pages/studyView/StudyViewPage.tsx

```diff
@@ -105,14 +109,10 @@ export default class StudyViewPage extends React.Component<IStudyViewPageProps,
             () => props.routing.location.query,
             query => {
                 if ('studyId' in query) {
-                    this.store.studyIds = (query.studyId as string).split(",");
+                    this.store.queriedIdsStr = (query.studyId as string);
```

Comment:
always check to see if the id is different before setting it.  mobx will fire even when you set a property with the same value and that will results in unnecessary re-render

Reply:
will update it once https://github.com/cBioPortal/cbioportal-frontend/pull/1458 is merged